### ISSUE TITLE
Weird options asked after `lando start`ing a platform.sh recipe project

### ISSUE THREAD

**Issue by thasmo** [lando/lando#2612]
**TL/DR**
When running `lando init` when setting up a `platform.sh` project before setting up the actual `platform.sh` project (e.g by running `platform project:set-remote ...` in the project directory) the behavior described below happens.

---

When running `lando start` and after the `platform.sh` CLI got installed within that process, two options need to be provided:
* Source directory (default: .):
* Build destination (default: _www):

After `lando start` has finished there's a new symlink `_www/` which points to the `public/` directory, which is the actual webroot.
It seems something is triggering some kind of setup/install routine which creates this symlink?

The `.lando.yml` file was created running `lando init` selecting `current directory` and the `platform.sh` recipe. I had to provide an access token which enabled me to select the desired platform.sh project.

**Steps**
```bash
git clone <platform.sh project>
lando init
lando start
```

**Terminal Excerpt**
```bash
To use the Platform.sh CLI, run:
platform
    Source directory (default: .):
    Build destination (default: _www):
Building application app (runtime type: php:7.4)
Found a composer.json file; installing dependencies
```

**.lando.yml**
```yaml
name: <projectname>
recipe: platformsh
config:
  id: <platformsh-identifier>
```

**.platform.app.yaml**
```yaml
name: app

type: php:7.4
build:
    flavor: composer

variables:
    env:
        APP_ENV: 'prod'
        APP_DEBUG: 0
        TYPO3_CONTEXT: 'Production'

runtime:
    extensions:
        - apcu
        - imagick
    
relationships:
    database: 'mysqldb:mysql'
    rediscache: 'rediscache:redis'

disk: 3072

mounts:
    'var/cache':
        source: local
        source_path: var/cache
    'var/charset':
        source: local
        source_path: var/charset
    'var/lock':
        source: local
        source_path: var/lock
    'var/log':
        source: local
        source_path: var/log
    'var/session':
        source: local
        source_path: var/session
    'var/transient':
        source: local
        source_path: var/transient
    'public/fileadmin':
        source: local
        source_path: fileadmin
    'public/uploads':
        source: local
        source_path: uploads
    'public/typo3temp':
        source: local
        source_path: typo3temp
    'public/typo3conf/writeable':
        source: local
        source_path: typo3-config

web:
    locations:
        '/':
            root: 'public'
            passthru: '/index.php'
            index:
                - 'index.php'
            allow: false
            rules:
                '\.(jpe?g|png|gif|svgz?|css|js|map|ico|bmp|eot|woff2?|otf|ttf)$':
                    expires: 7d
                    allow: true
                '^/robots\.txt$':
                    allow: true
                '^/sitemap\.xml$':
                    allow: true
                '/Sitemap\.xsl$':
                    allow: true
                '^/google3927d432d1e8f4f4\.html$':
                    allow: true
                '^/BingSiteAuth\.xml$':
                    allow: true
            headers: &headerControls
                X-Content-Type-Options: nosniff
                X-Frame-Options: SAMEORIGIN
                X-XSS-Protection: 1; mode=block
        '/static':
            root: 'public/static'
            expires: 1y
            scripts: false
            allow: true
            passthru: false
            headers:
                Service-Worker-Allowed: '/'
                X-Content-Type-Options: nosniff
            rules:
                '\.webmanifest$':
                    headers:
                        <<: *headerControls
                        Content-Type: application/manifest+json
        '/fileadmin':
            root: 'public/fileadmin'
            scripts: false
            allow: true
            passthru: '/index.php'
            headers: *headerControls
        '/fileadmin/_processed_':
            root: 'public/fileadmin/_processed_'
            expires: 1y
            scripts: false
            allow: true
            passthru: false
            headers: *headerControls
        '/fileadmin/csv':
            root: 'public/fileadmin/csv'
            expires: 1y
            scripts: false
            allow: true
            passthru: false
            headers: *headerControls
        '/fileadmin/domain-verification':
            root: 'public/fileadmin/domain-verification'
            expires: 1y
            scripts: false
            allow: true
            passthru: false
            headers: *headerControls
        '/fileadmin/form':
            root: 'public/fileadmin/form'
            expires: 1y
            scripts: false
            allow: true
            passthru: false
            headers: *headerControls
        '/fileadmin/handwerkprofis':
            root: 'public/fileadmin/handwerkprofis'
            expires: 1y
            scripts: false
            allow: true
            passthru: false
            headers: *headerControls
        '/fileadmin/sg_cookie_optin':
            root: 'public/fileadmin/sg_cookie_optin'
            expires: 1y
            scripts: true
            allow: true
            passthru: false
            headers: *headerControls
        '/uploads':
            root: 'public/uploads'
            scripts: false
            allow: true
            passthru: '/index.php'
            headers: *headerControls
        '/typo3temp/assets':
            root: 'public/typo3temp/assets'
            expires: 7d
            scripts: false
            allow: true
            passthru: false
            rules:
                '\.js\.gzip$':
                    headers:
                        <<: *headerControls
                        Content-Type: text/javascript
                        Content-Encoding: gzip
                '\.css\.gzip$':
                    headers:
                        <<: *headerControls
                        Content-Type: text/css
                        Content-Encoding: gzip
        '/typo3conf/LocalConfiguration.php':
            allow: false
        '/typo3conf/AdditionalConfiguration.php':
            allow: false

hooks:
    build: |
        if [ -s .nvmrc ]; then
            unset NPM_CONFIG_PREFIX
            curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.2/install.sh | dash
            export NVM_DIR="$HOME/.nvm"
            [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
            nvm install --latest-npm
        fi

        set -ex

        bash install-redis.sh 5.2.2

        if [ -s package.json ] && [ -s package-lock.json ]; then
            npm ci
            npm run build:production
            rm -rf node_modules/
        fi

        cd public/typo3conf/
        ln -s writeable/ENABLE_INSTALL_TOOL ENABLE_INSTALL_TOOL
    deploy: |
        set -x

        bin/console cache:clear --no-warmup || rm -rf var/cache/*
        bin/console cache:warmup
        vendor/bin/typo3cms cache:flushgroups system

        touch public/typo3conf/ENABLE_INSTALL_TOOL
        vendor/bin/typo3cms database:updateschema safe

        vendor/bin/typo3cms cache:flushgroups pages
    post_deploy: |
        set -x

        vendor/bin/typo3cms install:fixfolderstructure

        vendor/bin/typo3cms upgrade:prepare
        vendor/bin/typo3cms upgrade:run all

crons:
    typo3:
        spec: '*/5 * * * *'
        cmd: 'vendor/bin/typo3cms scheduler:run'
```

**Environment**
```
Lando v3.0.12
Windows 10.0.19041 Build 19041
Docker for Windows 2.3.0.5 (48029)
Docker Engine 19.03.12
Hyper-V 10.0.19041.488
Hyper-V Manager 10.0.19041.1
WSL2 Ubuntu 20.04
```

**Comment by pirog** [comment]
@thasmo you are going to have to give us way more detail here. At the bare minimum we need to know the lando version and probably the contents of all your platform.sh config files.

**Comment by thasmo** [comment]
Pardon, of course. I'm going to update the issue.

**Comment by pirog** [comment]
@thasmo knowing the steps you took before `lando start` would be good as well eg did you `lando init` your platform site or did you `git clone && lando start` etc.

**Comment by thasmo** [comment]
Indeed. Will be updating with more information continously asap.

**Comment by thasmo** [comment]
Updated it with some information; also with this:
```
After lando start has finished there's a new symlink _www/ which points to the public/ directory, which is the actual webroot.
It seems something is triggering some kind of setup/install routine which creates this symlink?
```

**Comment by pirog** [comment]
@thasmo the symlink is currently expected, the interactive questions asking for source and build directories is not eg `lando start` should not be blocking. 

Under the hood the `platform` cli is running `platform local:build APPNAME` to mimic the build that happens on platform.sh itself.

**Comment by thasmo** [comment]
@pirog I figured I did not yet run `platform project:set-remote <project-id>` after `git clone <project>` and that seems to be the issue. After running `platform project:set-remote <project-id>` the `platform local:build` seems to determine the `source` and `build` directories somehow and does not ask for it. Going to play around with it some more and will add more information if I find anything.

**Comment by pirog** [comment]
ahhh interesting @thasmo that is very helpful. Maybe there is a way for us to set that explicitly.

**Comment by thasmo** [comment]
@pirog, one thing `platform project:set-remote ...` does is to set the git remote for platform.sh in the local `.git/config` file:

```ini
[remote "platform"]
	url = <project-id>@git.eu-4.platform.sh:<project-id>.git
	fetch = +refs/heads/*:refs/remotes/platform/*
```

But besides that I don't know how the `platform.sh` CLI does relate the local project directory to the actual `platform.sh` project; maybe it can determine it solely via the configured git remote.

In general I thought running `lando init` for a `platform.sh` project was a bit odd, because I had to provide an access token only to select the `platform.sh` project from a list.

**Comment by thasmo** [comment]
@pirog, if the local project directory is already set up for `platform.sh`, it is possible to get the project id by running `platform project:info id`; that would yield a better developer experience than having to create an access token just to scroll through dozens of projects to set the id for lando.

**Comment by thasmo** [comment]
By the way, also have a look at ` platform project:list` and ` platform project:list --help` which could help listing projects without the need for a separate access token.

**Comment by pirog** [comment]
The access token is actually pretty important so I’m not sure that part of
the flow is going to change.

There is a lot more happening than just grabbing the ID particularly if you
are asking lando to git clone the project for you.
